Objects/Converters/ConverterCSI/ConverterCSIShared/Models/Element1DAnalyticalResultConverter.cs

```diff
@@ -0,0 +1,331 @@
+    }
+
+    private ICollection<ResultSet1D> GetAnalysisResultsForFrame(Element1D element1D)
+    {
+      int forcesSuccess = -1;
+
+      // Reference variables for CSI API
+      int numberOfResults = 0;
+      string[] obj,
+        elm,
+        loadCase,
+        stepType;
+      obj = elm = loadCase = stepType = Array.Empty<string>();
+      double[] objSta,
+        elmSta,
+        stepNum,
+        p,
+        v2,
+        v3,
+        t,
+        m2,
+        m3;
+      objSta = elmSta = stepNum = p = v2 = v3 = t = m2 = m3 = Array.Empty<double>();
+
+      if (SendForces(element1D.type))
+      {
+        forcesSuccess = sapModel.Results.FrameForce(
+          element1D.name,
+          eItemTypeElm.ObjectElm,
+          ref numberOfResults,
+          ref obj,
+          ref objSta,
+          ref elm,
+          ref elmSta,
+          ref loadCase,
+          ref stepType,
+          ref stepNum,
+          ref p,
+          ref v2,
+          ref v3,
+          ref t,
+          ref m2,
+          ref m3
+        );
+      }
+      else
+      {
+        return new List<ResultSet1D>();
+      }
+
+      // Value used to normalized output station of forces between 0 and 1
+      var lengthOf1dElement = objSta.Max();
+
+      return CreateLoadCombinationResults(
+        element1D,
+        forcesSuccess,
+        numberOfResults,
+        null,
+        loadCase,
+        (int i) => (float)(objSta[i] / lengthOf1dElement),
+        p,
+        v2,
+        v3,
+        t,
+        m2,
+        m3);
+    }
+    private ResultSet1D GetOrCreateResult(Dictionary<string, ResultSet1D> dict, string loadCaseName)
+    {
+      if (!dict.TryGetValue(loadCaseName, out ResultSet1D comboResults))
+      {
+        Base loadCaseOrCombination = loadCombinationsAndCases[loadCaseName];
+        comboResults = new ResultSet1D(new())
+        {
+          resultCase = loadCaseOrCombination
+        };
+        dict[loadCaseName] = comboResults;
+      }
+      return comboResults;
+    }
+    private ICollection<ResultSet1D> GetAnalysisResultsForPier(Element1D element1D)
+    {
+      int forcesSuccess = -1;
+
+      // Reference variables for CSI API
+      int numberOfResults = 0;
+      string[] storyName,
+        pierName,
+        loadCase,
+        location;
+      storyName = pierName = loadCase = location = new string[1];
+      double[] p,
+        v2,
+        v3,
+        t,
+        m2,
+        m3;
+      p = v2 = v3 = t = m2 = m3 = new double[1];
+
+      if (SendForces(element1D.type))
+      {
+        forcesSuccess = sapModel.Results.PierForce(
+          ref numberOfResults,
+          ref storyName,
+          ref pierName,
+          ref loadCase,
+          ref location,
+          ref p,
+          ref v2,
+          ref v3,
+          ref t,
+          ref m2,
+          ref m3
+        );
+      }
+
+      return CreateLoadCombinationResults(
```

Comment:
if we're in control of this function signature, passing a struct might be cleaner, it can also do the initialisation in one place and declutter where we are making these calls...

Reply:
> if we're in control of this function signature, passing a struct might be cleaner

Yeah I thought the same thing, but the way that I'm picturing it would turn this
```
return CreateLoadCombinationResults(
  elementName,
  forcesSuccess,
  numberOfResults,
  null,
  loadCase,
  (int i) => (float)(objSta[i] / lengthOf1dElement),
  p,
  v2,
  v3,
  t,
  m2,
  m3);
```
into this
```
DTO dto = new(
  elementName,
  forcesSuccess,
  numberOfResults,
  null,
  loadCase,
  (int i) => (float)(objSta[i] / lengthOf1dElement),
  p,
  v2,
  v3,
  t,
  m2,
  m3);

return CreateLoadCombinationResults(dto);
```
which really isn't any cleaner. But let me know if you were picturing something different.